### PR TITLE
[TACHYON-379]Restructure netty server code

### PR DESCRIPTION
**Pull request by viper-kun**
If we create netty client, there will be some common function as netty server.First, we get common function and move them into ThreadFactoryUtils file. Next, I will provider netty client.

### CODE REVIEW

**Comment by haoyuan** [comment]
@viper-kun  Could you please add the description of the PR? Also, create a JIRA for it. Thanks!

More info can be found here: http://tachyon-project.org/master/Contributing-to-Tachyon.html

**Comment by calvinjia** [comment]
@viper-kun I agree with the idea, but I think these utils belong outside of the general thread factory utils class and in a netty specific utils class.

**Comment by viper-kun** [comment]
@calvinjia I agree with you. I will add netty utils class and move common class into it.

**Comment by haoyuan** [comment]
Why change `deploy/vagrant/run_vb.sh` file mode?

**Comment by uronce-cc** [comment]
@haoyuan the file's mode is changed from 766 to 644, I don't know why that's necessary :)

**Comment by calvinjia** [comment]
I think that file needs to have the execute bit set, since the user will run that script to launch instances. It should not be 644.

**Comment by viper-kun** [comment]
@haoyuan  It is not necessary to change file mode.

File: core/src/main/java/tachyon/util/NettyUtils.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+      String threadPrefix) {
+    ThreadFactory threadFactory = ThreadFactoryUtils.build(threadPrefix);
+
+    switch (type) {
+      case NIO:
+        return new NioEventLoopGroup(numThreads, threadFactory);
+      case EPOLL:
+        return new EpollEventLoopGroup(numThreads, threadFactory);
+      default:
+        throw new IllegalArgumentException("Unknown io mode: " + type);
```
Comment (rxin) [inline core/src/main/java/tachyon/util/NettyUtils.java]: I think you forgot to change the error message when you copied this code from Spark.

**Comment by calvinjia** [comment]
+1 LGTM

**Pull request merged**